examples/mistral_qlora_finetune_example.py

```diff
@@ -0,0 +1,54 @@
+"""Example of how to fine-tune a model using the QLoraTrainer class."""
+
+import os
+
+from datasets import load_from_disk
+
+from prompt2model.model_trainer.qlora_trainer import QLoraTrainer
+from prompt2model.utils.dataset_utils import format_train_data, make_combined_datasets
+
+if __name__ == "__main__":
+    # load in datasets
+    retrieved_dataset_dict = load_from_disk("demo_retrieved_dataset_dict")
+    retrieved_dataset = retrieved_dataset_dict["train"]
+    generated_dataset = load_from_disk("demo_generated_dataset")
+    dataset_list = [retrieved_dataset, generated_dataset]
+
+    # combine datasets and create train and eval splits
+    train_dataset = make_combined_datasets(dataset_list)
+    splits = train_dataset.train_test_split(test_size=0.1)
+    train_dataset = splits["train"]  # has 2 cols: "input_col" and "output_col"
+    eval_dataset = splits["test"]  # has 2 cols: "input_col" and "output_col"
+    formatted_train_dataset = format_train_data(
+        train_dataset
+    )  # combined into one col: "text"
+    formatted_eval_dataset = format_train_data(
+        eval_dataset
+    )  # combined into one col: "text"
```

Comment:
These data structure "type hints" could probably be split into two standalone comments: one for train_dataset/eval_dataset and one for their formatted counterparts, with additional explanation for why these two copies are necessary